### DISCUSSION TITLE
Proposal: Make `WalletConnect Cloud` optional features non-blocking for `Web3Modal` wallet connection logic

### DISCUSSION THREAD

**Discussion by SergeyVolynkin** [WalletConnect/walletconnect-monorepo#3719] [Developer Support]
Hi, we recently encountered an issue in production that some of our Hong Kong production customers were getting
 `Error: No internet connection detected. Please restart your network and try again.`
- ![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/34238697/9b68b898-48be-4078-bd6f-42433d1a950d)

Which blocked them from signing into the system with their MetaMask(chrome plugin)

Which is `P0` for our production system as if you can not sign in, you can not withdraw funds through UI (they still can withdraw from the chain directly, but it is much more of a hustle).

Proposal(cc @glitch-txs @0xAsimetriq 🙇‍♂️):
- To decouple logic of signing in via non-QR methods (plugin, iOS app installed on same device), decouple from the other nice-to-have WalletConnect features (i.e. metrics, QR Code sign in);
- Or at least provide the option to proxy all WalletConnect requests through other AWS cloud. In case of hybrid DApps (where both cloud and chain interaction exists), that will make sure it's either both WalletConnect and some other AWS Cloud requests are going through or they both are blocked… ("All or nothing" is a better experience for our production use case, I believe, at least it is easier to debug)

3rd Option we're considering:
- Implement MetaMask(and other most common wallets) connectors as an additional escape hatch if WalletConnect Cloud goes down. We would want only to support WalletConnect in our system, for simplicity; But given if WalletConnect Cloud goes down, our Auth goes down, this is a serious `P0` risk for our production system.

**Comment by SergeyVolynkin** [comment]
cc @luccamordente for visibility

**Comment by SergeyVolynkin** [comment]
(cc @glitch-txs @0xAsimetriq 🙇‍♂️)

**Comment by xzilja** [comment]
We are planning improvements around offline mode and better handling of errors like this, part of it can be allowing user to continue with the modal via other means of connection like metamask as you mentioned.

**Comment by glitch-txs** [comment]
Hi @SergeyVolynkin have you tried this with Web3Modal v3? I'm not able to reproduce this there

**Reply by SergeyVolynkin** [reply]
Hi @glitch-txs, will be looking into our Sentry logs and testing myself this upcoming week, will circle back

**Reply by SergeyVolynkin** [reply]
@glitch-txs we'll be prod-releasing WalletConnect v3 this week, + it will take me 1 week to collect metrics on Sentry (if all good, will be closing this discussion as resolved in 2 weeks)

**Reply by SergeyVolynkin** [reply]
@0xAsimetriq, @glitch-txs 

we released WalletConnect v3 on our prod
```
    "@web3modal/wagmi": "^3.0.2",
    "viem": "^1.16.1",
    "wagmi": "^1.4.3"
```

Sentry still keeps raising issues that significant % of prod customers(>10%) having the issue
<img width="1144" alt="Screenshot 2023-10-21 at 1 58 36 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/34238697/e0260f11-824f-4766-bae7-0e0c8620365e">


> Re:  @0xAsimetriq  We are planning improvements around offline mode and better handling of errors like this

Sorry, @0xAsimetriq, do you know what is the target release version for the improvements? (we'll also try to update to the latest on our side continuously)

**Reply by SergeyVolynkin** [reply]
Hi @0xAsimetriq @glitch-txs, just a quick follow up that this error seems to be resolved, at least error rate for us is down to 0:
<img width="318" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/34238697/e7ef01ba-90b4-4b3e-aebc-860a42f184d1">

**Comment by SanLeo461** [comment]
Had a user report this issue, using @rainbow-me/rainbowkit 1.1.1
